src/main/scala/devices/tilelink/Error.scala

```diff
@@ -10,15 +10,18 @@ import freechips.rocketchip.util._
 import scala.math.min
 
 /** Adds a /dev/null slave that generates TL error response messages */
-class TLError(params: DevNullParams, beatBytes: Int = 4)(implicit p: Parameters)
-    extends DevNullDevice(params, beatBytes, new SimpleDevice("error-device", Seq("sifive,error0")))
+class TLError(params: DevNullParams, buffer: Boolean = true, beatBytes: Int = 4)(implicit p: Parameters)
+    extends DevNullDevice(params,
+      minLatency = if (buffer) 1 else 0,
+      beatBytes, new SimpleDevice("error-device", Seq("sifive,error0")))
 {
   lazy val module = new LazyModuleImp(this) {
     import TLMessages._
     import TLPermissions._
 
     val (in, edge) = node.in(0)
-    val a = Queue(in.a, 1)
+    val a = if (buffer) {Queue(in.a, 1)} else in.a
```

Comment:
Any reason you did not do this also for in.c?

Reply:
hmm, good question. because my device in question had no in.c. You're right, i should add